blockchain/chain_sync/src/chain_muxer.rs

```diff
@@ -165,7 +165,7 @@ where
     M: Provider + Sync + Send + 'static,
     C: Consensus,
 {
-    #[allow(clippy::too_many_arguments)]
+    #[allow(clippy::too_many_arguments, clippy::result_large_err)]
```

Comment:
Would it be too awkward to box the errors?

Reply:
This doesn't work

Reply:
Please explain why.

Reply:
it worked out

Reply:
If it worked out then do we need this suppression?

Reply:
It worked out partly, meaning if we were ok with `map_err`. But David wanted to get of them completely (AFAI understood) and this was not possible (as it propagated all over the codebase). So for now we decided to leave the attribute.

Reply:
I'm 90% sure we can get it to work without having to change too much code. I'll open a new issue for that.

Reply:
Sounds good